thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/formatter/DTOFormatter.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.apache.pinot.thirdeye.detection.formatter;
+
+import java.util.Map;
+import org.apache.pinot.thirdeye.datalayer.dto.AbstractDTO;
+
+
+/**
+ * Class for building UI formatters for DTOs. Takes in a DTO and returns
+ * a map that contains human-readable data for display on the UI.
+ */
+public interface DTOFormatter<T extends AbstractDTO> {
```

Comment:
If it is a general purpose formatter, would be better to put into a separate package.

Reply:
Good idea. Will move to thirdeye.formatter package.

Reply:
Agree. I am writing a parameter spec.